Gatekeeper

pinecat opened the pull request:
PR for testing out Github Kanban (migrating from Trello).

pinecat committed:
gatekeeper: added handler and struct to config

Started working on the gatekeeper functionality.  Added struct for
gatekeeper options in config.go.  Also added the gatekeeper handler, and
registered it with the other bot handlers.  Finally, added fix for
privileged intents from the nickname-tracking branch.  Although, it was
slightly changed so that we are not overwriting a ton of values by
updating with a struct.  Instead we update value individually.

pinecat committed:
gatekeeper: remove extranious newline

Removed an uneeded newline from the gatekeeper handler function.

pinecat committed:
gatekeeper: handler, config, and new command

Added the update command, used to force the bot to update various
settings.  Currently, it is being used for the 'update gatekeeper'
command to update the welcome channel with new gatekeeper settings from
the config file.

Messed around with some things in cfg.go, but ultimately ended up
keeping everything the same for now.

Added a new handler but then it broke, so it is all commented out for
now.  Will work on fixing it.

pinecat committed:
gatekeeper: logic in place

Added the logic for the gatekeeper to the router.  I tried adding it in
gatekeeper.go again, as its own handler, but it is still is giving me a
ton of errors.  Adding the logic to the router, however, everything
works perfectly.

The gatekeeper portion of the router config is now being updated as well
from the config we read in when an 'update gk' command is used.

pinecat committed:
gatekeeper: role add on join

Users now receive the role specified by the role id in the gatekeeper
config when joining the server.

pinecat committed:
gatekeeper: basic functionality

Got the basic functionality of the gatekeeper working.  At the moment,
the logic which handles the gatekeeper is just baked into the router.
As such, there is no way to disable it.  Although, it can probably,
technically be disabled by just not having any gatekeeper values in your
config file.

Also updated the parameter names in the gatekeeper config to better
reflect what they represent (i.e. Channel is now ChannelID).

pinecat committed:
Merge branch 'master' into gatekeeper

pinecat committed:
gatekeeper: remove debug line

Removed logging line for debugging.  Add to TODO to figure out why
putting the gatekeeper logic in its own handler isn't working for me.

pinecat committed:
gatekeeper: debug line removed

Removed logging line used for debugging.  Added TODO to figure out why
putting the gatekeeper logic in its own handler is not working on my
computer.

cebarks committed:
Fix api 404 bug when checking user permissions

Also moves handler out of Router handler and back into (handlers.)GatekeeperMsgHandler()

cebarks committed:
remove debug code

cebarks committed:
move 'update' command to under 'debug'A

Also adds:
- a few helper functions in cmd.Context
- force reload of config from disk
- Default handler for commands will not print out all help commands help
text (just like the 'commands' command)

cebarks reviewed (approved):
I push a commit to fix the weird bug we were getting and revert handler code to its own function instead of router.

pinecat committed:
gatekeeper: move gatekeeper functionality

Move the gatekeeper logic to its own handler.  Finally got the config
reading properly, so we are not getting a nil GuildID.  I just moved the
action events to the new handler.

pinecat committed:
gatekeeper: forgot to save

Forgot to save the file I needed to commit last commit.
